extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultExtension.java

```diff
@@ -18,31 +19,40 @@
 import org.eclipse.edc.runtime.metamodel.annotation.Inject;
 import org.eclipse.edc.runtime.metamodel.annotation.Provider;
 import org.eclipse.edc.runtime.metamodel.annotation.Provides;
+import org.eclipse.edc.spi.EdcException;
 import org.eclipse.edc.spi.http.EdcHttpClient;
+import org.eclipse.edc.spi.monitor.Monitor;
 import org.eclipse.edc.spi.security.CertificateResolver;
 import org.eclipse.edc.spi.security.PrivateKeyResolver;
 import org.eclipse.edc.spi.security.Vault;
 import org.eclipse.edc.spi.security.VaultPrivateKeyResolver;
+import org.eclipse.edc.spi.system.ExecutorInstrumentation;
 import org.eclipse.edc.spi.system.ServiceExtension;
 import org.eclipse.edc.spi.system.ServiceExtensionContext;
 import org.eclipse.edc.spi.types.TypeManager;
 
-@Provides({ Vault.class, PrivateKeyResolver.class, CertificateResolver.class })
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+
+@Provides({Vault.class, PrivateKeyResolver.class, CertificateResolver.class, HashicorpVaultClient.class})
 @Extension(value = HashicorpVaultExtension.NAME)
 public class HashicorpVaultExtension implements ServiceExtension {
-
-
     public static final String NAME = "Hashicorp Vault";
 
-
     @Inject
     private EdcHttpClient httpClient;
 
     @Inject
     private TypeManager typeManager;
 
+    @Inject
+    private ExecutorInstrumentation executorInstrumentation;
+
     private Vault vault;
     private PrivateKeyResolver privateKeyResolver;
+    private ScheduledExecutorService scheduledExecutorService;
+    private HashicorpVaultClient hashicorpVaultClient;
+    private boolean isTokenRenewable;
```

Comment:
this is information that the extension does not need. it should be handled inside the vault client.

Reply:
Shouldn't the extension take care of initalizing/starting the client properly since it also has to take care of shutting it down properly?